textFilter.js

```diff
@@ -1,188 +1,69 @@
+// selects the elements in dom as variables
 const searchInput = document.querySelector('.input-search-keyword');
 const roomsUl = document.querySelector('.rooms');
-
-
-
-
-
-    function search(){
-       
-
-        const searchString = searchInput.value;
-       
-        //if string empty display all rooms
-        if(searchString == ''){
-            renderAll();
-        
-        }
-        //if string is not empty render rooms matching the string
-        else{
-            render(searchInput.value);
+// Decides if the input is empty or not
+function startOfSearch(){
+    const preSearchString = searchInput.value;
+    // if preSearchString/input is empty display all rooms
+    if(preSearchString == ''){
+        renderAll;
+    }
+    //if string is not empty render rooms matching the string
+    else{
+        renderWFilter(preSearchString);
+    }
+}
+// renders all rooms if the input (searchString) is empty
+async function renderAll(){
+    const data = await APIGet();
+    roomsUl.innerText = '';
+    data.challenges.forEach(challenge => {
+        const roomsUl = document.querySelector(".rooms");
+        const item = document.createElement("li");
+        const content = document.createElement("p");
+
+        content.innerText = "\nTitle: " + challenge.title + "\nDescription: " + challenge.description;
```

Comment:
Använd interpolation (det är oftast enklare)
ex. `Title ${challenge.title}`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Good idea, we can use that for the proper render Vedad made if he hasn't already or if he did it in another way.